pkg/controller/account/account_controller.go

```diff
@@ -285,19 +285,22 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	var byocRoleID string
 
-	// If the account is BYOC, needs some different set up
+	// If the account is BYOC, override awsSetupClient with creds from byoc CR
 	if currentAcctInstance.Spec.BYOC {
 		reqLogger.Info("BYOC account")
-		if currentAcctInstance.Status.State == "" || currentAcctInstance.Status.Claimed != true {
-			// Create client for BYOC account
-			byocAWSClient, accountClaim, err := r.getBYOCClient(currentAcctInstance)
-			if err != nil {
-				if accountClaim != nil {
-					r.accountClaimBYOCError(reqLogger, accountClaim, err)
-				}
-				return reconcile.Result{}, err
+		// Create client for BYOC account
+		byocSetupClient, accountClaim, err := r.getBYOCClient(currentAcctInstance)
+		if err != nil {
+			if accountClaim != nil {
+				r.accountClaimBYOCError(reqLogger, accountClaim, err)
 			}
+			return reconcile.Result{}, err
+		}
 
+		// re-assign the client to byoc instantiation
+		awsSetupClient = byocSetupClient
```

Comment:
Can we just update the `awsSetupClient` reference to `byocSetupClient` everywhere?

Reply:
I would think not, the byoc client is created once the account_claim is identified as a BYOC type. Only then, do we reassign `byocSetupClient` to `awsSetupClient` so the rest of the code can remain unchanged.